modules/javafx.graphics/src/main/java/javafx/scene/CssStyleHelper.java

```diff
@@ -176,7 +177,7 @@ static CssStyleHelper createStyleHelper(final Node node) {
 
         helper.cacheContainer = new CacheContainer(node, styleMap, depth);
 
-        helper.firstStyleableAncestor = findFirstStyleableAncestor(node);
+        helper.firstStyleableAncestor = new WeakReference<>(findFirstStyleableAncestor(node));
```

Comment:
Can you investigate for an alternative to set `firstStyleableAncestor` to null when the related `node` is removed from scenegraph
or
may be `null` the reference to `Node.styleHelper` itself when Node is removed from scenegraph. This will result in recreating the `Node.styleHelper` next time when it is added back to scenegraph.